pkg/deploy/deployer.go

```diff
@@ -146,3 +151,37 @@ func (d *deployer) getParameters(ps map[string]interface{}) *arm.Parameters {
 
 	return parameters
 }
+
+func (d *deployer) removeFailedScaleset(ctx context.Context, resourceGroupName, vmssName string) (deleted bool) {
+	// Check if scaleset exists. If not, no need to delete it.
+	vmss, err := d.vmss.Get(ctx, resourceGroupName, vmssName)
+	if isNotFound(err) {
+		return true
+	}
+
+	// If it exists and is in failed state, try deleting it so naming conflict doesn't occur during retry.
+	if *vmss.ProvisioningState == string(mgmtcompute.ProvisioningStateFailed) {
+		d.log.Printf("deleting failed scaleset %s", *vmss.Name)
+		err = d.vmss.DeleteAndWait(ctx, resourceGroupName, *vmss.Name)
+		if err != nil {
+			d.log.Print(err)
+			return false
+		}
+	}
+
+	timeoutErr := wait.PollImmediate(time.Second, 10*time.Minute, func() (bool, error) {
+		_, err := d.vmss.Get(ctx, resourceGroupName, vmssName)
+		if isNotFound(err) {
+			return true, nil
+		}
+		return false, nil
+	})
+	return timeoutErr == nil // If it didn't time out, it was deleted.
```

Comment:
I know that I suggested to use `wait.PollImmediate`, but looking at the bigger picture now I'm wondering if we actually need it. I think `d.vmss.DeleteAndWait` does wait for the VMSS to be actually deleted. Or am I wrong?

Reply:
Yeah, DeleteAndWait should be enough. It's pointless to double-check that it's gone.

Reply:
removed the redundant check